feat: Support regex usage in Azure Service Bus scaler. (#917)
kedacore/keda-docs: content/docs/2.9/scalers/azure-service-bus.md

- `subscriptionName` - Name of the Azure Service Bus queue to scale on. (Optional*, Required when `topicName` is specified)
- `namespace` - Name of the Azure Service Bus namespace that contains your queue or topic. (Optional*, Required when pod identity is used)
- `connectionFromEnv` - Name of the environment variable your deployment uses to get the connection string of the Azure Service Bus namespace. (Optional)

> - `useRegex` - This parameter allows the usage of regex (in the `queueName` or `subscriptionName` parameters). (Values: `boolean`, Default: `false`, Optional)

Reviewer: I'd say that in other places we have this
```suggestion
- `useRegex` - This parameter allows the usage of regex (in the `queueName` or `subscriptionName` parameters). (Values: `true`, `false`, Default: `false`, Optional)
```

Author: I'll change this. Any other changes besides this?

> - `useRegex` - This parameter allows the usage of regex (in the `queueName` or `subscriptionName` parameters). (Values: `true`, `false`, Default: `false`, Optional)

Reviewer 2: ```suggestion
- `useRegex` - Provides indication whether or not a regex is used in the `queueName` or `subscriptionName` parameters. (Values: `true`, `false`, Default: `false`, Optional)
```

> - `operation` - Operation that will be used to compute the number of messages when `useRegex` is set to `true`. (Values: `sum`, `max`, or `avg`, Default: `sum`, Optional).

Reviewer 2: ```suggestion
- `operation` - Defines how to compute the number of messages across entities when `useRegex` is set to `true`. (Values: `sum`, `max`, or `avg`, Default: `sum`, Optional).
```

> - `operation` - Operation that will be used to compute the number of messages when `useRegex` is set to `true`. Required when `useRegex` is `true`. (Values: `sum`, `max`, or `avg`, Default: `sum`, Optional).

Reviewer 2: why not just use entityNameRegex or so?

Reviewer 2: I think it's safe to assume that when the regex field was provided, that a regex should be used

Author: I aligned this with the way RabbitMQ supports regex, but what you're suggesting is also fair.

Reviewer 2: @JorTurFer @zroubalik wdyt?

Reviewer: it's true that in RabbitMQ we have `useRegex` as a boolean and `queueName` but there we don't have 2 different parameters (`queueName` and `topicName`) only one (`queueName`). Maybe in this context we can use `entityNameRegex` as single field for both.
BTW, isn't `topicName` instead of `subscriptionName` or I'm wrong?

Author: I didn't want to have a field entity name regex because then we would need another field to determine which entity it is (queue or subscription). 

Although, I guess if it's for a queue the metadata will only contain `entityNameRegex`, while if it's for a subscription, the metadata will contain `topicName` and `entityNameRegex`.

Author: The presence of a `queueName`, or `subscriptionName` already tells us what entity it is.

Reviewer: are we going to apply the regex in the subscription name instead of the topic name?

Author: Yes, there can be multiple subscriptions within a topic, so this will make it easier to specify multiple of them.
And that was also the original ask - https://github.com/kedacore/keda/issues/1624

Reviewer: oh, thanks for the clarification. Once again, the problem is that I'm a noobie with Azure Service Bus xD
